js/components/geolocation.js

```diff
@@ -0,0 +1,175 @@
+Fliplet.FormBuilder.field('geolocation', {
+  name: 'Geolocation',
+  category: 'Advanced',
+  props: {
+    value: {
+      type: String
+    },
+    description: {
+      type: String
+    },
+    autofill: {
+      type: Boolean,
+      default: false
+    },
+    preciseLocationRequired: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data: function() {
+    return {
+      firstTimeSaved: false,
+      isLoading: false,
+      buttonClicked: false,
+      accuracy: null
+    };
+  },
+  validations: function() {
+    var rules = {
+      value: {}
+    };
+
+    if (this.required && !this.readonly) {
+      rules.value.required = window.validators.required;
+    }
+
+    return rules;
+  },
+  mounted: function() {
+    if (this.autofill) {
+      this.getLocation();
+    }
+  },
+  created: function() {
+    Fliplet.FormBuilder.on('reset', this.onReset);
+    Fliplet.Hooks.on('beforeFormSubmit', this.onBeforeSubmit);
+  },
+  destroyed: function() {
+    Fliplet.FormBuilder.off('reset', this.onReset);
+    Fliplet.Hooks.off('beforeFormSubmit', this.onBeforeSubmit);
+  },
+  methods: {
+    setValue: function(result) {
+      return `${result.coords.latitude}, ${result.coords.longitude}`;
+    },
+    getLocation: function() {
+      var $vm = this;
+
+      $vm.buttonClicked = true;
+      $vm.isLoading = true;
+
+      var location = Fliplet.Navigator.location({
```

Comment:
I would turn

```
Fliplet.Navigator.location({
        maximumAge: 0,
        timeout: 30000,
        enableHighAccuracy: true
      })
```

into a `getDeviceLocation` function so that the API options can be centralised in one place